core/src/main/scala/com/salesforce/op/stages/impl/selector/SelectedCombiner.scala

```diff
@@ -0,0 +1,245 @@
+          bestModelUID = summary1.bestModelUID + " " + summary2.bestModelUID,
+          bestModelName = summary1.bestModelName + " " + summary2.bestModelName,
+          bestModelType = summary1.bestModelType + " " + summary2.bestModelType,
+          validationResults = summary1.validationResults ++ summary2.validationResults,
+          trainEvaluation = evaluate(model.transform(dataset)),
+          holdoutEvaluation = None
+        )
+        setMetadata(summary.toMetadata().toSummaryMetadata())
+      }
+    }
+
+    val (metricValue1, metricValue2) = (getMet(metricValueOpt1), getMet(metricValueOpt2))
+
+    val strategy = getCombinationStrategy()
+    val (weight1, weight2) = strategy match {
+      case CombinationStrategy.Best =>
+        if (metricValue1 > metricValue2) (1.0, 0.0) else (0.0, 1.0)
+      case CombinationStrategy.Weighted =>
+        (metricValue1 / (metricValue1 + metricValue2), metricValue2 / (metricValue1 + metricValue2))
+      case CombinationStrategy.Equal =>
+        (0.5, 0.5)
+      case s => throw new RuntimeException(s"Combination strategy $s is not supported")
+    }
+
+    val model: SelectedCombinerModel = new SelectedCombinerModel(
+      weight1 = weight1, weight2 = weight2, strategy = strategy,
+      operationName = operationName, uid = uid
+    )
+      .setEvaluators(evaluators)
+      .setParent(this)
+      .setInput(in1.asFeatureLike[RealNN], in2.asFeatureLike[Prediction], in3.asFeatureLike[Prediction])
+      .setOutputFeatureName(getOutputFeatureName)
+
+    makeMeta(model)
+    model.setMetadata(getMetadata())
+  }
+
+}
+
+final class SelectedCombinerModel private[op]
+(
+  val weight1: Double,
+  val weight2: Double,
+  val strategy: CombinationStrategy,
+  val operationName: String,
+  val uid: String
+)(
+  implicit val tti1: TypeTag[RealNN],
+  val tti2: TypeTag[Prediction],
+  val tti3: TypeTag[Prediction],
+  val tto: TypeTag[Prediction],
+  val ttov: TypeTag[Prediction#Value]
+) extends Model[SelectedCombinerModel] with OpTransformer3[RealNN, Prediction, Prediction, Prediction]
+  with HasTestEval {
+
+  override protected[op] def outputsColNamesMap: Map[String, String] =
+    Map(ModelSelectorNames.outputParamName -> getOutputFeatureName)
+
+  override def transformFn: (RealNN, Prediction, Prediction) => Prediction = (_, p1: Prediction, p2: Prediction) => {
+    val rawPrediction = p1.rawPrediction.zip(p2.rawPrediction).map{ case (v1, v2) => v1 * weight1 + v2 * weight2 }
+    val probability = p1.probability.zip(p2.probability).map{ case (v1, v2) => v1 * weight1 + v2 * weight2 }
+    val prediction =
+      if (probability.nonEmpty) probability.indexOf(probability.max).toDouble
+      else p1.prediction * weight1 + p2.prediction * weight2
+    Prediction(prediction = prediction, probability = probability, rawPrediction = rawPrediction)
+  }
+
+  lazy val labelColName: String = in1.name
+
+  @transient private var evaluatorList: Seq[OpEvaluatorBase[_ <: EvaluationMetrics]] = Seq.empty
```

Comment:
Why does this need to be transient?

Reply:
so that we can save the model

Reply:
So no we do not need a custom serializer - evaluators do not need to be serialized